Wikipedia/Code/WebViewController.m

```diff
@@ -303,7 +303,8 @@ - (void)handleLateJavascriptTransformScriptMessage:(NSString *)messageString {
         [self.webView wmf_setPageProtected:!self.article.editable];
     } else if ([messageString isEqualToString:@"addFooterContainer"]) {
         [self.webView wmf_addFooterContainer];
-    } else if ([messageString isEqualToString:@"addFooterReadMore"]) {
+        // ns of articles with Read More pages equals 0.
+    } else if ([messageString isEqualToString:@"addFooterReadMore"] && self.article.ns == 0) {
```

Comment:
should also check for `!self.article.isMain` - could add a computed readonly property `hasReadMore` to `MWKArticle` to use here and in `WMFArticleViewController`